Source/Scene/ArcGisMapServerImageryProvider.js

```diff
@@ -297,6 +301,91 @@ define([
+            vertical = CesiumMath.toDegrees(latitude);
+            sr = '4326';
+        } else {
+            var projected = imageryProvider._tilingScheme.projection.project(new Cartographic(longitude, latitude, 0.0));
+            horizontal = projected.x;
+            vertical = projected.y;
+            sr = '3857';
+        }
+
+        var url = imageryProvider._url + '/identify?f=json&tolerance=2&geometryType=esriGeometryPoint';
+        url += '&geometry=' + horizontal + ',' + vertical;
+        url += '&mapExtent=' + rectangle.west + ',' + rectangle.south + ',' + rectangle.east + ',' + rectangle.north;
+        url += '&imageDisplay=' + imageryProvider._tileWidth + ',' + imageryProvider._tileHeight + ',96';
+        url += '&sr=' + sr;
+
+        url += '&layers=visible';
+        if (defined(imageryProvider._layers)) {
+            url += ':' + imageryProvider._layers;
+        }
+
+        if (defined(imageryProvider._token)) {
+            url += '&token=' + imageryProvider._token;
+        }
+
+        if (defined(imageryProvider._proxy)) {
+            url = imageryProvider._proxy.getURL(url);
+        }
+
+        return url;
+    }
+
+    function jsonToFeatures(json) {
+        var result = [];
+
+        var features = json.results;
+        if (!defined(features)) {
+            return result;
+        }
+
+        for (var i = 0; i < features.length; ++i) {
+            var feature = features[i];
+
+            var featureInfo = new ImageryLayerFeatureInfo();
+            featureInfo.data = feature;
+            featureInfo.name = feature.value;
+            featureInfo.properties = feature.attributes;
+            featureInfo.configureDescriptionFromProperties(feature.attributes);
+
+            // If this is a point feature, use the coordinates of the point.
+            if (feature.geometryType === 'esriGeometryPoint' && feature.geometry) {
+                var wkid = feature.geometry.spatialReference && feature.geometry.spatialReference.wkid ? feature.geometry.spatialReference.wkid : 4326;
+                if (wkid === 4326 || wkid === 4283) {
+                    featureInfo.position = Cartographic.fromDegrees(feature.geometry.x, feature.geometry.y, feature.geometry.z);
+                } else if (wkid === 102100 || wkid === 900913 || wkid === 3857) {
+                    var projection = new WebMercatorProjection();
+                    featureInfo.position = projection.unproject(new Cartesian3(feature.geometry.x, feature.geometry.y, feature.geometry.z));
+                }
+            }
+
+            result.push(featureInfo);
+        }
+
+        return result;
+    }
+
+    function updateToken(imageryProvider) {
+        if (!defined(imageryProvider._newTokenRequestInFlight) && defined(imageryProvider._requestNewToken)) {
+            imageryProvider._newTokenRequestInFlight = imageryProvider._requestNewToken().then(function(newToken) {
+                delete imageryProvider._newTokenRequestInFlight;
```

Comment:
Set it undefined, don't delete it.

Reply:
No worries...but why? just trying to learn why this is bad style here.

Thanks.